components/Map.js

```diff
@@ -26,10 +26,14 @@ export default class Map extends React.Component {
         longitudeDelta: 0.0421,
       },
     };
-    this.onRegionChange = this.onRegionChange.bind(this);
+
```

Comment:
Adding debounce in constructor() is common to avoid creating a new function on each render call.